base/src/multiply.cu

```diff
@@ -455,6 +454,7 @@ void blockDiaCsrMultiplyKernelDiaProps_4x4( const IndexType *row_offsets,
     // Iterate over the rows of the matrix. One warp per two rows.
     for ( int aRowId = blockIdx.x * nHalfWarps + halfWarpId ; aRowId < num_block_rows ; aRowId += gridDim.x * nHalfWarps )
     {
+        unsigned int active_mask = utils::activemask();

```

Comment:
So IIUC Masking is required as this loop is over half warps. Is __activemask the right choice here rather than a ballot or some other approach to checking if only the lower half is cooperating?

Reply:
I think this is not guaranteed to work, but happens to work on all current hardware. I would be more comfortable if this were to use `cooperative_groups::` instead, but it might make sense to improve our testing a bit before doing that kind of change.

Reply:
If I remember correctly, the changes here were made to follow the pattern from `multicolor_ilu_solver.cu`